### ISSUE TITLE
newrelic_synthetics_secure_credential randomly with error "Error: entity not found, or found more than one"

### ISSUE THREAD

**Issue by inno-ng** [newrelic/terraform-provider-newrelic#1961]
**newrelic_synthetics_secure_credential** randomly with error "Error: entity not found, or found more than one" even though there are no creds in that account. 

Same thing happens when sometimes it first creates creds and then on rerun again throws same error again. 

### Terraform Version
```Terraform v1.1.9
on linux_amd64
```

Provider config:

```
provider "newrelic" {
  account_id = "1111111"
  api_key    = "NRAK-xxxxxxxxxxxx"
  region     = "US"
}


terraform {
  required_providers {
    newrelic = {
      source  = "newrelic/newrelic"
      version = "3.0.1"
    }
  }
}
```

### Affected Resource(s)
- `newrelic_synthetics_secure_credential`
- `newrelic_synthetics_script_monitor`

How to reproduce:

create a new relic secure credentials using terraform. 
Delete it from new relic and re run the terraform again. it will throw 

```
❯ terraform apply
newrelic_synthetics_secure_credential.test: Refreshing state... [id=TEST_BASE_URL]
╷
│ Error: entity not found, or found more than one
│ 
│   with newrelic_synthetics_secure_credential.test,
│   on main.tf line 16, in resource "newrelic_synthetics_secure_credential" "test":
│   16: resource "newrelic_synthetics_secure_credential" "test" {
│ 
╵
❯ terraform destroy
newrelic_synthetics_secure_credential.test: Refreshing state... [id=TEST_BASE_URL]
╷
│ Error: entity not found, or found more than one
│ 
│   with newrelic_synthetics_secure_credential.test,
│   on main.tf line 16, in resource "newrelic_synthetics_secure_credential" "test":
│   16: resource "newrelic_synthetics_secure_credential" "test" {
```

Code was working fine with 2.49.1 but since we have updated to 3.0.1 due to our requirement of using node 16 for script monitor, credentials setup is not working what so ever. 

another issue: **newrelic_synthetics_script_monitor** and **newrelic_synthetics_monitor** uses **locations_public** value which is referenced in "**https://docs.newrelic.com/docs/synthetics/synthetic-monitoring/administration/synthetic-public-minion-ips/.**" but value is wrong in this doc. for e.g **AWS_US_EAST_1** is **US_EAST_1**. Please check this and make updates.

**Comment by shiva-bandari** [comment]
Hi @inno-ng,
Thanks for raising the issue.
We are able to reproduce, we will prioritise this issue.

**Comment by mbazhlekova** [comment]
@inno-ng We released a fix for this issue with version 3.4.3.

**Comment by khrahul** [comment]
` Error: strconv.ParseInt: parsing "null": invalid syntax
│ 
│   with newrelic_synthetics_secure_credential.test["password"],
│   on main.tf line 79, in resource "newrelic_synthetics_secure_credential" "test":
│   79: resource "newrelic_synthetics_secure_credential" "test" {
│ `

if we run pipeline again , instead of modifying/ignore the previously created secure credential , terraform apply exits with this error. while terraform plan works fine. seems like an GO API error.
@shiva-bandari
